src/lib/components/dualTimeView.svelte

```diff
@@ -36,36 +39,88 @@
         return formattedTime ? `${formattedTime} ago` : 'Just now';
     }
 
+    function shouldHidePopover(hideTooltip: () => void) {
+        let isMouseOverTooltip = false;
+
+        if (!tooltipElement) {
+            hideTooltip();
+            return;
+        }
+
+        if (tooltipElement) {
+            tooltipElement.addEventListener('mouseenter', () => {
+                isMouseOverTooltip = true;
+            });
+
+            tooltipElement.addEventListener('mouseleave', () => {
+                isMouseOverTooltip = false;
+                hideTooltip();
+            });
```

Comment:
these event listeners are added multiple times I think. maybe do `on:mouseenter` on the element itself 👍🏻